Use route URI in EXCLUDE rules

lucianobargmann opened the pull request:
This will allow to remove Telescope routes from documentation.

shalvah commented:
Hmmm. I'm not sure we should do this. Aren't Telescope routes named?

shalvah commented:
Also, if we do this for `exclude`, we have to do it for `include` too.

lucianobargmann commented:
I believe it is already in the "Include" - that is why I am proposing the update.

Check out line 73:
 && str_is($routeRule['match']['prefixes'] ?? [], $route->uri())

And the complete include function:
private function shouldIncludeRoute(Route $route, array $routeRule, array $mustIncludes, bool $usingDingoRouter)
    {
        $matchesVersion = $usingDingoRouter
            ? ! empty(array_intersect($route->versions(), $routeRule['match']['versions'] ?? []))
            : true;
        return str_is($mustIncludes, $route->getName())
            || (str_is($routeRule['match']['domains'] ?? [], $route->getDomain())
            && str_is($routeRule['match']['prefixes'] ?? [], $route->uri())
            && $matchesVersion);
    }

shalvah commented:
It's not in `include`. The rule you pointed out is the `match.prefixes` rule (see https://laravel-apidoc-generator.readthedocs.io/en/latest/config.html#prefixes). I'm talking about the `include` section (the counterpart to `exclude` (https://laravel-apidoc-generator.readthedocs.io/en/latest/config.html#include-and-exclude)

lucianobargmann commented:
Got it @shalvah - How would you suggest me to remove Telescope from generated docs? May we add it to Readme?

Thanks for your time

shalvah commented:
@lucianobargmann Are the Telescope routes named?

shalvah commented:
@lucianobargmann I see the Telescope routes are apparently not made. https://github.com/laravel/telescope/blob/d7a3ca4338b1a1527b8d19a5d62015a468f8ca56/src/Http/routes.php
DOn't know why they aren't. However, I'll accept this change in the interim. You need to implement this for `include` too, as well as update the documentation to specify that the values in `exclude` and `include` are first treated as route names, then as URLs.

lucianobargmann commented:
Wait!

This is how I solved it, using the current version:
I added a prefix to my routes and changed config/apidoc.php accordingly:

'prefixes' => [
                    'api/*'
                ],

Thank you very much for your time looking into this PR.

Obrigado,

*LUCIANO BARGMANN*


On Fri, May 17, 2019 at 3:43 PM Shalvah <notifications@github.com> wrote:

> @lucianobargmann <https://github.com/lucianobargmann> I see the Telescope
> routes are apparently not made.
> https://github.com/laravel/telescope/blob/d7a3ca4338b1a1527b8d19a5d62015a468f8ca56/src/Http/routes.php
> DOn't know why they aren't. However, I'll accept this change in the
> interim. You need to implement this for include too, as well as update
> the documentation to specify that the values in exclude and include are
> first treated as route names, then as URLs.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mpociot/laravel-apidoc-generator/pull/507?email_source=notifications&email_token=AADZB3C3UUZY5VYZGMP56U3PV34DJA5CNFSM4HLFNBE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVVRR4Y#issuecomment-493558003>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADZB3HN6N2OIVHZSQ3Y3W3PV34DJANCNFSM4HLFNBEQ>
> .
>

shalvah commented:
Okay. Going ahead to include this anyway.
